src/transformStateWithClones.js

```diff
@@ -6,8 +6,30 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let step = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      step = {};
+    }
+
+    if (action.type === 'addProperties') {
+      Object.assign(step, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete step[key];
+      }
+    }
```

Comment:
Instead of multiple `if` statements, use a `switch` statement to handle different action types. This will make the code more organized and easier to read, and it will ensure that each action type is mutually exclusive. Remember to include a `default` case to handle any unexpected action types.